### PR TITLE
refactor pickers

### LINKED ISSUE #58: Color chips resize slightly when selecting item
I made a pretty rad [rainbow theme](https://mozilla.github.io/ThemesRFun/?theme=XQAAAAIJAQAAAAAAAABBKYkm0ypxtqu9mB2KK4d4ccIgevwats_sOw9RE2e7oEmWHhMAwySPjd9PWLIRa3i2_IhUrLinugIGwsoHSMQuNXKNe1-IgpFgRFbLo1UG2TF55MYKTtoclNxtrMJBetGlpKbwobZcLVIFPkwaOkSdavp8b6ppcZlVRUNrXgQQ1Cx3FMM3YZAmug4ll-C1byd2tikv6wnjhFt1CFa0B4re5ZesBEdgLxIZxFgnGo5R9OfmbEJpfgM___P2OWw) but noticed that when I select an item to customize, the text grow comically large (FF Nightly), causing the text to wrap and sometimes the circular color chip will squish (see Yellow, Green, Blue, and Purple chips).

themesrnifty

### PR DESCRIPTION
**Pull request by johngruen**
This changes around the picker ui fairly drastically, but i think it's a lot cooler/more direct this way. I'd like to merge it to see how it feels. Either way, extracting the picker css out a bit makes it easier to work with the picker and background selectors going forward.

Fixes #58

### CODE REVIEW

File: src/web/lib/components/BrowserPreview/index.js [hunk 1]
```diff
@@ -35,9 +35,9 @@ export const BrowserPreview = ({
 
   const tabGenerator = () => {
     const tabTitles = ['Tab One', 'Tab Two'];
-    if (size === 'large') {
-      tabTitles.push('Tab Three');
-    }
+    // if (size === 'large') {
```
Comment (johngruen) [inline src/web/lib/components/BrowserPreview/index.js]: I could probably get rid of these

File: src/web/lib/components/ThemeBackgroundPicker/index.js [hunk 2]
```diff
@@ -1,43 +1,70 @@
...
+
+  handleClickOutside() {
+    this.setState({ selected: false });
+  }
+
+  render() {
+    const { theme, setBackground } = this.props;
+    const { selected } = this.state;
+    const accentcolor = colorToCSS(theme.colors.accentcolor);
+    // Note: app initializes with no bg set so we have to check before adding this straight to CSS
```
Comment (johngruen) [inline src/web/lib/components/ThemeBackgroundPicker/index.js]: i should clarify that i mean in the default theme

**Comment by johngruen** [comment]
Here's a screenshot:
image

File: src/web/lib/components/ThemeColorsEditor/index.js [hunk 3]
```diff
@@ -29,7 +30,11 @@ export const UndoRedoButtons = ({
   </div>
 );
 
-export default class ThemeColorsEditor extends React.Component {
+class ThemeColorsEditor extends React.Component {
+  handleClickOutside() {
+    this.props.setSelectedColor('none');
```
Comment (lmorchard) [inline src/web/lib/components/ThemeColorsEditor/index.js]: This action creator expects an object, should be more like:
```javascript
this.props.setSelectedColor({ name: null });
```

File: src/web/lib/components/ThemeColorsEditor/index.js [hunk 4]
```diff
@@ -44,34 +49,33 @@ export default class ThemeColorsEditor extends React.Component {
...
-                  color={{ h: color.h, s: color.s, l: color.l, a: color.a * 0.01 }}
-                  disableAlpha={!colorsWithAlpha.includes(name)}
-                  onChangeComplete={({ hsl: { h, s, l, a } }) =>
-                    setColor({ name, h, s: s * 100, l: l * 100, a: a * 100 })}
-                />
-              </dd>
+                <span className="color__swatch"style={{ backgroundColor: colorToCSS(color) }} />
+                <span className="color__label">{colorLabels[name]}</span>
+                <span className="color__picker">
+                  <BlockPicker
```
Comment (lmorchard) [inline src/web/lib/components/ThemeColorsEditor/index.js]: Personal thing, but I don't like this color picker. It's pretty restrictive - no alpha and I can't pick any colors I like.

Comment (johngruen) [reply src/web/lib/components/ThemeColorsEditor/index.js]: I'm not thrilled about it either, but it had a caret so i'm using it for the time being

**Comment by pdehaan** [comment]
Will this get auto-deployed to https://mozilla.github.io/ThemesRFun/, or is that something that needs to be done manually?

**Comment by lmorchard** [comment]
Manually deployment from `development` branch right now - see #53

**Pull request merged**